lib/pure/osproc.nim

```diff
@@ -525,10 +527,17 @@ when defined(Windows) and not defined(useNimRtl):
       handle: Handle
       atTheEnd: bool
 
+  proc closeHandleCheck(handle: Handle) {.inline.} =
+    if handle.closeHandle() == 0:
+      raiseOSError(osLastError())
+
+  proc fileClose[T: Handle | FileHandle](h: var T) {.inline.} =
+    if h > 4:
```

Comment:
@demotomohiro pre-existing but where does this magic `h > 4` comes from? shouldn't that at least be h > 2 ?

eg assuming stdin=0,stdout=1,stderr=2, what about files with fileno equal to 3 or 4 ?

at very least we need a comment explanining where this comes from